### ISSUE TITLE
WebView2 (version 117) crashes on Windows 11 Store version with Desktop Bridge technology

### ISSUE THREAD

**Issue by sadfklsdslk** [MicrosoftEdge/WebView2Feedback#3803]
**Description**
I encountered a crash issue with WebView2. This issue only occurs on the Windows 11 Store version, with apps using the Desktop Bridge technology. The problem is in WebView2 version 117, where WebView2 cannot be launched, and it seems to crash in the sandbox. Please refer to the attached stack trace.

**Version**
SDK: 117.0.2045
OS: Win11

![image](https://github.com/MicrosoftEdge/WebView2Feedback/assets/145662360/f5a08ac3-0b1b-4f73-8972-b49caa6ce47c)

[dump.zip](https://github.com/MicrosoftEdge/WebView2Feedback/files/12682957/dump.zip)

[AB#46707609](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/46707609)

**Comment by champnic** [comment]
Hey @sadfklsdslk - Thanks for the bug report, and sorry you are hitting this. Was this working before the 117 runtime? And can you tell me what framework your app is built with, like WPF, Winforms, UWP, etc.?

**Comment by sadfklsdslk** [comment]
> Hey @sadfklsdslk - Thanks for the bug report, and sorry you are hitting this. Was this working before the 117 runtime? And can you tell me what framework your app is built with, like WPF, Winforms, UWP, etc.?

The issue was not present before the 117 version, as far as we know. According to our internal statistics, the problem has surged in the last few days. Our application is a Win32 C++ program.

**Comment by champnic** [comment]
Do you have a sample app or link to the store app that we can use to try and reproduce this issue?

**Comment by vbryh-msft** [comment]
Additionally to what Nic is asking - could you please share installer logs from the recently affected devices? - [instructions what to collect](https://github.com/MicrosoftEdge/WebView2Feedback/blob/main/diagnostics/install.md)

**Comment by vbryh-msft** [comment]
@sadfklsdslk we have not heard from you - was you able to solve the issue? Do you have any logs for us to look into?

**Comment by sadfklsdslk** [comment]
> @sadfklsdslk we have not heard from you - was you able to solve the issue? Do you have any logs for us to look into?

The latest version of Webview2 (118.0.2088.61) still has this issue (in the Win11 app store). I tried using -- no sanbox to close the sandbox, but it still didn't work. I have now provided more log information. I need some time to create a separate entity demo package (appx), and now I will provide some log information. Could you please take a look at these log information first.

[118.0.2088.61_log.zip](https://github.com/MicrosoftEdge/WebView2Feedback/files/13172632/118.0.2088.61_log.zip)

**Comment by sadfklsdslk** [comment]
> Do you have a sample app or link to the store app that we can use to try and reproduce this issue? And can you confirm you are using the evergreen WebView2 runtime install, and not including the fixed version runtime binaries with your app deployment?

I have provided more log information, please take a look again. The probability of this problem occurring in Win11 is 100%. Our online store version (appx) package no longer dares to turn on Webview2, but we can add a whitelist for experimentation. If I need to create a separate entity demo package (appx), please give me some time

**Comment by vbryh-msft** [comment]
the error from the dump is `ERROR_NEEDS_REMEDIATION | The application cannot be started. Try reinstalling the application to fix the problem.` - and nothing suspicious in installer logs. Does Edge browser work for those devices?

**Comment by sadfklsdslk** [comment]
> the error from the dump is `ERROR_NEEDS_REMEDIATION | The application cannot be started. Try reinstalling the application to fix the problem.` - and nothing suspicious in installer logs. Does Edge browser work for those devices?

As long as it's not an APPX or running in a store, there's no problem. And the computer is a brand new Win11 Edge browser, which is running normally. I will provide an appx demo for you to see later

**Comment by vbryh-msft** [comment]
Hi - do you have an app with repro; do you still need help with this?

**Comment by sadfklsdslk** [comment]
> Hi - do you have an app with repro; do you still need help with this?

OK, I need help. Currently, this phenomenon can still be replicated. Win11 store version Webview2 crashes, it must occur. No matter what the error is, webview2 should not crash. Now I will attach this appx package to you. It requires an official signature from Microsoft before it can be installed properly. If you need a signature, please let me know and I will go to the store to sign it.
https://drive.weixin.qq.com/s?k=AJEAIQdfAAoOH1OqwV

**Comment by sadfklsdslk** [comment]
> Hi - do you have an app with repro; do you still need help with this?

I don't know why the same code, Webview2 can run normally in the Win10 store, but Webview2 will crash in the Win11 store. Please help me. The same appx installation package. If you need a signature, please let me know and I will sign it because it requires Microsoft review and can be quite troublesome

**Comment by sadfklsdslk** [comment]
> Hi - do you have an app with repro; do you still need help with this?

I solved this problem by closing the sandbox. The Win11 store version requires the sandbox to be closed.

`options->put_AdditionalBrowserArguments(L"--no-sandbox");`

Win10 store version does not require it, I hope this information is helpful for you to fix bugs. Currently, we have enabled users to run webview2 normally through this means

**Comment by vbryh-msft** [comment]
@sadfklsdslk thank you for the update - glad you made it work.

**Comment by vbryh-msft** [comment]
@sadfklsdslk  - could you please share from where is information that "The Win11 store version requires the sandbox to be closed"?

**Comment by sadfklsdslk** [comment]
> @sadfklsdslk - could you please share from where is information that "The Win11 store version requires the sandbox to be closed"?

No, I discovered it myself. I found that closing the sandbox has an effect and does not crash. This may be related to the fact that the Win11 store version comes with Microsoft's built-in sandbox
